### PR TITLE
Show caasp node versions and indicate if nodes have updates available

### LINKED ISSUE #450: Show CaaSP Node version and whether updates are pending on the summary view

### PR DESCRIPTION
**Pull request by nwmac**
fixes #450

### CODE REVIEW

File: src/frontend/packages/suse-extensions/src/custom/kubernetes/services/kubernetes-endpoint.service.ts [hunk 1]
```diff
@@ -65,6 +80,50 @@ export class KubernetesEndpointService {
...
+        info.version = Object.keys(versions).join(', ');
+        info.versionMismatch = Object.keys(versions).length !== 1;
+        return info;
+      })
+    )
+  }
+
+  // Check for the specified annotation with a value other than 'no'
+  private hasBooleanAnnoation(annotations: Annotations, annotation: string): number {
+    return annotations[annotation] && annotations[annotation] === 'no' ? 0 : 1
```
Comment (richard-cox) [inline src/frontend/packages/suse-extensions/src/custom/kubernetes/services/kubernetes-endpoint.service.ts]: I tried this with a node that only had the caasp version annotation and it came up with 1 node needing upgrade with a disruptive and a security update. In cases where it's missing should this return 0?

**Review by richard-cox — CHANGES_REQUESTED** [review]
I made a couple of tweaks whilst reviewing
- securityUpdates looks at `CAASP_SECURITY_UPDATES_ANNOTATION` value
- fixed a typo in a function name

There's one question inline regarding behaviour on missing annotations

I think we should also add it to the below places as well. The user will want to investigate and find which nodes have the issues. If so I can have a look in a separate PR?
- conditionally add info in the nodes list if the endpoint type is caasp
- add the info to the node summary page

File: src/frontend/packages/suse-extensions/src/custom/kubernetes/services/kubernetes-endpoint.service.ts (hunk 1 above)
Comment (nwmac) [reply src/frontend/packages/suse-extensions/src/custom/kubernetes/services/kubernetes-endpoint.service.ts]: Logic broken

Comment (nwmac) [reply src/frontend/packages/suse-extensions/src/custom/kubernetes/services/kubernetes-endpoint.service.ts]: return !annotations[annotation] || annotations[annotation] && annotations[annotation] === 'no' ? 0 : 1

**Pull request merged**